decoder/pokemon.go

```diff
@@ -145,11 +145,119 @@ func getPokemonRecord(ctx context.Context, db db.DbDetails, encounterId string)
 	return &pokemon, nil
 }
 
+// hasChangesPokemon compares two Pokemon structs
+// Ignore Pvp field
+// Float tolerance for Lat, Lon
 func hasChangesPokemon(old *Pokemon, new *Pokemon) bool {
-	return !cmp.Equal(old, new, cmp.Options{
-		ignoreNearFloats,
-		cmpopts.IgnoreFields(Pokemon{}, "Pvp"),
-	})
+	if old.Id != new.Id {
```

Comment:
I am glad you included the comment to ignore the pvp field; this characteristic could get lost. 
There are some code generators available for golang which write code for you instead of using reflection; perhaps we should investigate one (though we don't have soooo many objects it is impossible to check)
It would also be possible to sequence these checks to put the most likely changes first to save another few microseconds (lol)

Reply:
Yeah! I've been thinking about adding counters to check which ifs we hit the most, but I will do that once we'll get prometheus :joy:

Yap, tried to make all required notes in parts we might touch in the future, so those methods won't get missed.